### PR TITLE
app: boards: add configuration for the mks_canable_v20 board

### PR DESCRIPTION
**Pull request by KozhinovAlexander**
Add configuration for the mks_canable_v20 board.

### CODE REVIEW

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# The three lines below are needed to reduce ROM usage
+CONFIG_LTO=y
+CONFIG_SIZE_OPTIMIZATIONS=y
```
Comment (henrikbrixandersen) [inline app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf:3]: This is the default. Please remove `CONFIG_SIZE_OPTIMIZATIONS=y`.

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+CONFIG_ISR_TABLES_LOCAL_DECLARATION=y
+
+CONFIG_USB_DEVICE_GS_USB_TIMESTAMP=y
+CONFIG_USB_DEVICE_GS_USB_TERMINATION=n
```
Comment (henrikbrixandersen) [inline app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf:7]: These are automatically set based on the devicetree. Please remove these two lines.

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+CONFIG_USB_DEVICE_GS_USB_MAX_CHANNELS=1
```
Comment (henrikbrixandersen) [inline app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf:8]: This is the default. Please remove this line.

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+CONFIG_USB_DEVICE_GS_USB_TIMESTAMP_SOF=y
```
Comment (henrikbrixandersen) [inline app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf:9]: Please leave this up to the user to enable. SoF events are hard on the CPU usage and the cost of this added timestamp precision should be evaluated by the user based on their needs, not enabled by default on a board-by-board basis.

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+CONFIG_USB_DEVICE_GS_USB_IDENTIFICATION=y
```
Comment (henrikbrixandersen) [inline app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf:10]: This is the default. Please remove this line.

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+CONFIG_LOG=y
+CONFIG_USB_DRIVER_LOG_LEVEL_ERR=y
+CONFIG_USB_DEVICE_LOG_LEVEL_ERR=y
+CONFIG_USB_DEVICE_GS_USB_LOG_LEVEL_DBG=n
+CONFIG_CANNECTIVITY_LOG_LEVEL_DBG=n
```
Comment (henrikbrixandersen) [inline app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf:16]: Please remove these lines. Debug options should not be set on a board-by-board basis.

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+# Use candleLight_fw VID:PID:
+CONFIG_CANNECTIVITY_USB_VID=0x1d50
+CONFIG_CANNECTIVITY_USB_PID=0x606f
```
Comment (henrikbrixandersen) [inline app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf:20]: Please remove these lines. CANnectivity ships with its own USB VID:PID and should not use USB VID:PIDs allocated to another project/product.

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.overlay [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+};
+
+&zephyr_udc0 {
+	gs_usb0: gs_usb0 {
+		compatible = "gs_usb";
+	};
+};
+
+&timers2 {
+	/* Timer with 1us tick used for CAN timestamp */
```
Comment (henrikbrixandersen) [inline app/boards/makerbase/mks_canable_v20/mks_canable_v20.overlay:30]: Please remove this comment. The purpose is obvious.

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.overlay [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+	st,prescaler = <159>;
+	counters2: counter {
+		status = "okay";
+	};
+};
+
+&fdcan1 {
+	reg = <0x40006400 0x400>, <0x4000a400 0x450>;
+	reg-names = "m_can", "message_ram";
+	bosch,mram-cfg = <0x0 28 8 3 3 0 3 3>;
```
Comment (henrikbrixandersen) [inline app/boards/makerbase/mks_canable_v20/mks_canable_v20.overlay:40]: What is the rationale behind this customisation?

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.overlay [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+	status = "okay";
```
Comment (henrikbrixandersen) [inline app/boards/makerbase/mks_canable_v20/mks_canable_v20.overlay:41]: This is already marked as "okay" in the Zephyr board definition. No need for repeating it here. Please remove it.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

**Comment by henrikbrixandersen** [comment]
> A new affordable board supporting CANnectvity

Also, please change the Git commit log to something else. This board is not new, and "affordable" is very subjective.

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.overlay (hunk 8 above)
Comment (KozhinovAlexander) [reply app/boards/makerbase/mks_canable_v20/mks_canable_v20.overlay:30]: Could we please leave it. Reading this dts file first time does not give any clue, how prescaler is related to the tick value of 1us.  Just an example of typical determining of the tick value:

1. Without comment:

- Find PLL clock for timer (need to look in board's main device  tree)
- Divide PLL clock by prescaler + 1 (need to look up st's reference manual)
In total: x2 time consuming steps are needed

3. With comment:

- Take a look on comment and see value directly.

In total: x1 very simple step needed

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.overlay (hunk 9 above)
Comment (KozhinovAlexander) [reply app/boards/makerbase/mks_canable_v20/mks_canable_v20.overlay:40]: Good catch. I've forgot it during testing. Will remove.

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.overlay (hunk 10 above)
Comment (KozhinovAlexander) [reply app/boards/makerbase/mks_canable_v20/mks_canable_v20.overlay:41]: See comment above

**Comment by KozhinovAlexander** [comment]
> > A new affordable board supporting CANnectvity
> 
> Also, please change the Git commit log to something else. This board is not new, and "affordable" is very subjective.

Done. I've just followed similiar commits from your PR's for other boards now

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.overlay (hunk 8 above)
Comment (henrikbrixandersen) [reply app/boards/makerbase/mks_canable_v20/mks_canable_v20.overlay:30]: My point is, that the CANnectivity timestamp counter _always_ runs @ 1MHz. The comment does not add any value. 

Please remove it.

**Comment by KozhinovAlexander** [comment]
@henrikbrixandersen Thank you as always for very fast and confident review. I've made all the changes requested. Only one comment I would like to leave in place. Please take a look.

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.overlay (hunk 8 above)
Comment (KozhinovAlexander) [reply app/boards/makerbase/mks_canable_v20/mks_canable_v20.overlay:30]: Removed

**Comment by henrikbrixandersen** [comment]
@KozhinovAlexander Hey, please let the reviewer (in this case me) resolve the comments. It is up to the reviewer (not the author) to determine, if their review comment was resolved. 

This is good practice in general, not just in CANnectivity.

Thanks.

**Comment by henrikbrixandersen** [comment]
@KozhinovAlexander Hey, please let the reviewer (in this case me) resolve the comments. It is up to the reviewer (not the author) to determine, if their review comment was resolved. 

This is good practice in general, not just in CANnectivity.

Thanks.

**Comment by KozhinovAlexander** [comment]
> @KozhinovAlexander Hey, please let the reviewer (in this case me) resolve the comments. It is up to the reviewer (not the author) to determine, if their review comment was resolved.
> 
> This is good practice in general, not just in CANnectivity.
> 
> Thanks.

Oh. Sure. I see. Of course will do always from now on. Shall I unresolve then?

**Comment by henrikbrixandersen** [comment]
> Oh. Sure. I see. Of course will do always from now on. Shall I unresolve then?

No need, I unresolved them.

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf [hunk 11]
```diff
@@ -0,0 +1,3 @@
+# The three lines below are needed to reduce ROM usage
```
Comment (henrikbrixandersen) [inline app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf:1]:
```suggestion
# The two lines below are needed to reduce flash usage
```

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Looks good, thanks! One small nit:

File: app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf (hunk 11 above)
Comment (KozhinovAlexander) [reply app/boards/makerbase/mks_canable_v20/mks_canable_v20.conf]: Strage. I've applied the suggestion and github resolved this automatically. Thus it was not me 😄

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by KozhinovAlexander** [comment]
> Looks good, thanks! One small nit:

Thank you too a lot. Wish you merry christmas.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Wait, you've used the GitHub web UI for applying a fix. Please pull the branch, squash the two commit, and force-push.

**Comment by KozhinovAlexander** [comment]
> Wait, you've used the GitHub web UI for applying a fix. Please pull the branch, squash the two commits, and force-push.

Done

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
> > Wait, you've used the GitHub web UI for applying a fix. Please pull the branch, squash the two commits, and force-push.
> 
> Done

Thanks. Happy holidays!

**Pull request merged**